integration/connection.tf

```diff
@@ -340,6 +340,109 @@ resource "materialize_connection_postgres" "postgres_ssl_connection" {
   validate = false
 }
 
+resource "materialize_connection_mysql" "mysql_connection" {
+  name    = "mysql_connection"
+  comment = "connection mysql comment"
+
+  host = "mysql"
+  port = 3306
+  user {
+    text = "mysqluser"
+  }
+  password {
+    name          = materialize_secret.mysql_password.name
+    database_name = materialize_secret.mysql_password.database_name
+    schema_name   = materialize_secret.mysql_password.schema_name
+  }
+}
+
+resource "materialize_connection_mysql" "mysql_connection_aws_pl" {
+  name    = "mysql_connection_aws_pl"
+  comment = "connection mysql comment with aws privatelink"
+
+  host = "mysql"
+  port = 3306
+  user {
+    text = "mysqluser"
+  }
+  password {
+    name          = materialize_secret.mysql_password.name
+    database_name = materialize_secret.mysql_password.database_name
+    schema_name   = materialize_secret.mysql_password.schema_name
+  }
+
+  aws_privatelink {
+    name          = "privatelink_conn"
+    database_name = "materialize"
+    schema_name   = "public"
+  }
+  validate = false
+}
+
+resource "materialize_connection_mysql" "mysql_connection_with_secret" {
+  name = "mysql-connection-with-secret"
+  host = "mysql"
+  port = 3306
+  user {
+    secret {
+      name          = materialize_secret.mysql_user.name
+      database_name = materialize_secret.mysql_user.database_name
+      schema_name   = materialize_secret.mysql_user.schema_name
+    }
+  }
+  password {
+    name          = materialize_secret.mysql_password.name
+    database_name = materialize_secret.mysql_password.database_name
+    schema_name   = materialize_secret.mysql_password.schema_name
+  }
+  validate = false
+}
+
+resource "materialize_connection_mysql" "mysql_ssh_tunnel_connection" {
+  name = "mysql_ssh_tunnel_connection"
+  host = "mysql"
+  port = 3306
+  user {
+    text = "mysqluser"
+  }
+  password {
+    name          = materialize_secret.mysql_password.name
+    database_name = materialize_secret.mysql_password.database_name
+    schema_name   = materialize_secret.mysql_password.schema_name
+  }
+  ssh_tunnel {
+    name = materialize_connection_ssh_tunnel.ssh_connection.name
+  }
+  validate = false
+}
+
+resource "materialize_connection_mysql" "mysql_ssl_connection" {
+  name = "mysql_ssl_connection"
+  host = "mysql"
+  port = 3306
+  user {
+    text = "mysqluser"
+  }
+  password {
+    name          = materialize_secret.mysql_password.name
+    database_name = materialize_secret.mysql_password.database_name
+    schema_name   = materialize_secret.mysql_password.schema_name
+  }
+  ssl_mode = "required"
```

Comment:
this should probably be `verify-identity` since we're providing the server-ca and client-certs, we might as well enforce they are correct